hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/KeyManagerImpl.java

```diff
@@ -2205,6 +2272,167 @@ private void listStatusFindKeyInTableCache(
     return fileStatusList;
   }
 
+  public List<OzoneFileStatus> listStatusV1(OmKeyArgs args, boolean recursive,
+      String startKey, long numEntries, String clientAddress)
+          throws IOException {
+    Preconditions.checkNotNull(args, "Key args can not be null");
+
+    List<OzoneFileStatus> fileStatusList = new ArrayList<>();
+    if (numEntries <= 0) {
+      return fileStatusList;
+    }
+
+    String volumeName = args.getVolumeName();
+    String bucketName = args.getBucketName();
+    String keyName = args.getKeyName();
+    // A map sorted by OmKey to combine results from TableCache and DB.
+    TreeMap<String, OzoneFileStatus> cacheKeyMap = new TreeMap<>();
+    String seekKeyInDB = "";
+    long prefixKeyInDB = Long.MIN_VALUE;
+    String prefixPath = keyName;
+
+    metadataManager.getLock().acquireReadLock(BUCKET_LOCK, volumeName,
+            bucketName);
+    try {
+      if (Strings.isNullOrEmpty(startKey)) {
+        OzoneFileStatus fileStatus = getFileStatus(args, clientAddress);
+        if (fileStatus.isFile()) {
+          return Collections.singletonList(fileStatus);
+        }
+
+        // keyName is a directory
+        if (fileStatus.getKeyInfo() != null) {
+          seekKeyInDB = fileStatus.getKeyInfo().getObjectID()
+                  + OZONE_URI_DELIMITER;
+          prefixKeyInDB = fileStatus.getKeyInfo().getObjectID();
+        } else {
+          String bucketKey = metadataManager.getBucketKey(volumeName,
+                  bucketName);
+          OmBucketInfo omBucketInfo =
+                  metadataManager.getBucketTable().get(bucketKey);
+          seekKeyInDB = omBucketInfo.getObjectID()
+                  + OZONE_URI_DELIMITER;
+          prefixKeyInDB = omBucketInfo.getObjectID();
```

Comment:
Should reset prefixPath to bucketKey I think.

Reply:
"/vol/bucket/" is the root directory and the first level directories are under this. Thats the reason I left, `prefixPath` as empty. PrefixPath is used to build the absolute key path name which will be added to the OMKeyInfo#keyName. Existing code base don't have `volume/bucket name` in the key path name.

Reply:
Thanks for correct me. I am thinking more for this, here the case is that we cannot find the expected key info due to a non-exist key name, right. So maybe we should just return the empty list. Because we give a wrong key name.

Reply:
Yes, this condition is for the root directory listing. `fileSystem.listStatus(new Path("o3fs://bucket1.vol1/"))`
For a wrong or non-existing keyname, today it is throwing exception. I have referred code base [KeyManagerImpl.java#L1823](https://github.com/apache/hadoop-ozone/blob/master/hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/KeyManagerImpl.java#L1823)

Reply:
Okay, get it.